File: java/src/test/java/problems101_200/TriangleTest.java
package problems101_200;

import java.util.Arrays;
import java.util.stream.Collectors;
import org.junit.Test;
import sample.SampledTest;

import static com.google.common.truth.Truth.assertWithMessage;

public class TriangleTest extends SampledTest {
    @Test
    public void test() {
        for (Triangle solution : Triangle.values()) {
            for (TriangleSample sample : getSamples(TriangleSample[].class)) {
                assertWithMessage(sample.getMessage())
                    .that(
                        solution.minimumTotal(
                            Arrays
                                .stream(sample.input.triangle)
                                .map(ints ->
                                    Arrays
                                        .stream(ints)
                                        .boxed()
                                        .collect(Collectors.toList())
                                ).collect(Collectors.toList())
                        )
                    ).isEqualTo(sample.output);
            }
        }
    }
}

File: kotlin/src/test/kotlin/problems101_200/TriangleTest.kt
package problems101_200

import com.google.common.truth.Truth.assertWithMessage
import sample.SampledTest
import sample.component1
import sample.component2
import sample.component3
import sample.getSamples
import kotlin.test.Test

class TriangleTest : SampledTest() {
    @Test
    fun test() =
        Triangle.entries.forEach { solution ->
            getSamples<Array<TriangleSample>>().forEach { (input, output, message) ->
                assertWithMessage(message)
                    .that(solution.minimumTotal(input.triangle.map { it.toList() }))
                    .isEqualTo(output)
            }
        }
}
